hnn_core/cell.py

```diff
@@ -220,6 +224,52 @@ def __init__(self, L, diam, Ra, cm, end_pts=None):
     def __repr__(self):
         return f'L={self.L}, diam={self.diam}, cm={self.cm}, Ra={self.Ra}'
 
+    def __eq__(self, other):
+        if not isinstance(other, Section):
+            return NotImplemented
+
+        # Check equality for mechs
+        for mech_name in self.mechs.keys():
+            self_mech = self.mechs[mech_name]
+            other_mech = other.mechs[mech_name]
+            for attr in self_mech.keys():
+                if self_mech[attr] != other_mech[attr]:
+                    return False
+
+        # Check end_pts
+        for self_end_pt, other_end_pt in zip(self.end_pts, other.end_pts):
+            if self_end_pt != other_end_pt:
+                if np.testing.assert_almost_equal(self_end_pt,
```

Comment:
```suggestion
                if not np.testing.assert_almost_equal(self_end_pt,
```
?

Reply:
So I found out `np.testing.assert_almost_equal` returns `None` when the condition is true. `None` is treated as `false` in `if` condition